Develop

Petrovich-A opened the pull request.

Besvic commented on src/main/java/by/petrovich/tool/controller/EmployeePositionController.java:
```diff
@@ -0,0 +1,55 @@
+    @DeleteMapping("/{id}")
+    public ResponseEntity<String> delete(@PathVariable("id") Long id) {
+        employeePositionServiceImpl.delete(id);
+        return ResponseEntity.status(HttpStatus.OK).body("Employee position deleted successfully");
```
return id instead of string

Besvic commented on src/main/java/by/petrovich/tool/model/Employee.java:
```diff
@@ -0,0 +1,57 @@
+public class Employee {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(columnDefinition = "serial")
```
long is bigserial in postgresql

Besvic commented on src/main/java/by/petrovich/tool/model/Employee.java:
```diff
@@ -0,0 +1,57 @@
+@Entity
+public class Employee {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
```
sequance is better

Besvic commented on src/main/java/by/petrovich/tool/service/impl/EmployeePositionServiceImpl.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Override
+    public EmployeePositionResponseDto update(Long id, EmployeePositionRequestDto employeePositionRequestDto) {
+        Optional<EmployeePosition> employeePosition = positionRepository.findById(id);
```
for what? you can just call exist method and after that save all object

Petrovich-A replied:
I have to do the findById call anyway cause need convert toEntityUpdate with existing entity @MappingTarget EmployeePosition employeePosition in order to avoid the creating a new one

Besvic commented on src/main/java/by/petrovich/tool/service/impl/EmployeePositionServiceImpl.java:
```diff
@@ -0,0 +1,60 @@
+
+    @Override
+    public void delete(Long id) {
+        Optional<EmployeePosition> employeePosition = positionRepository.findById(id);
```
exist() too

Besvic commented on src/test/java/by/petrovich/tool/service/impl/EmployeePositionServiceImplTest.java:
```diff
@@ -0,0 +1,236 @@
+        List<EmployeePositionResponseDto> expected = Arrays.asList(employeePositionResponseDto1, employeePositionResponseDto2);
+
+        when(employeePositionMapper.toResponseDto(employeePosition1)).thenReturn(employeePositionResponseDto1);
+        when(employeePositionMapper.toResponseDto(employeePosition2)).thenReturn(employeePositionResponseDto2);
```
as i told earlier, you aren't need to write mocks for mappers.

Petrovich-A commented:
### Created a base *Spring* project:
- configured the **Postgresql** database
- implemented **liquibase** to create database and populate tables through the scripts in **yaml** format
- integrated API calls to fetch **EmployeePosition** data
- added project description to *README.md*
